File: 80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.c
int removeDuplicates(int* nums, int numsSize) {
    if (numsSize <= 2) return numsSize;
    
    int k = 2;  // Start from the third element
    
    for (int i = 2; i < numsSize; i++) {
        if (nums[i] != nums[k-2]) {
            nums[k] = nums[i];
            k++;
        }
    }
    
    return k;
}

File: 80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.go
func removeDuplicates(nums []int) int {
    if len(nums) <= 2 {
        return len(nums)
    }
    
    k := 2  // Start from the third element
    
    for i := 2; i < len(nums); i++ {
        if nums[i] != nums[k-2] {
            nums[k] = nums[i]
            k++
        }
    }
    
    return k
}